source/common/common/fancy_logger.cc

```diff
@@ -109,24 +141,87 @@ void FancyContext::initSink() {
   }
 }
 
-spdlog::logger* FancyContext::createLogger(std::string key, int level)
+spdlog::logger* FancyContext::createLogger(std::string key)
     ABSL_EXCLUSIVE_LOCKS_REQUIRED(fancy_log_lock_) {
   SpdLoggerSharedPtr new_logger =
       std::make_shared<spdlog::logger>(key, Logger::Registry::getSink());
   if (!Logger::Registry::getSink()->hasLock()) { // occurs in benchmark test
     initSink();
   }
-  level_enum lv = Logger::Context::getFancyDefaultLevel();
-  if (level > -1) {
-    lv = static_cast<level_enum>(level);
-  }
-  new_logger->set_level(lv);
+
+  new_logger->set_level(getLogLevel(key));
   new_logger->set_pattern(Logger::Context::getFancyLogFormat());
   new_logger->flush_on(level_enum::critical);
   fancy_log_map_->insert(std::make_pair(key, new_logger));
   return new_logger.get();
 }
 
+void FancyContext::updateVerbositySetting(std::vector<std::pair<absl::string_view, int>> updates) {
+  absl::WriterMutexLock ul(&fancy_log_lock_);
+  log_update_info_.clear();
+  for (const auto& [glob, level] : updates) {
+    if (level < 0 || level > 6) {
+      continue;
+    }
+    appendVerbosityLogUpdate(glob, static_cast<level_enum>(level));
+  }
+
+  for (auto& [key, logger] : *fancy_log_map_) {
```

Comment:
This will be O(Loggers * verbosity log updates)

It will also re-do some work; perhaps we can avoid getting / setting log level for something that wasn't updated.

Reply:
I am also looking into if there is a way to optimize it. 

This initial design is corresponding to the Google vmoudle behavior: it will override all previous updates whenever we get a new verbosity update, and support the file basename setting. Therefore at lease we need to check all the loggers during the verbosity update.

Let me know if you have any thoughts!

Reply:
We can go with it then for now since this is expected to rarely occur and we can revisit if needed.